adempiere/adempiere
# Add Support to Login with OpenID Connect

## Login with [Open ID Connect  ](https://openid.net/connect/)

This functionality allow login in Adempiere ZK OpenID Connect, specifically with this services:

- Github
- Google

## Step by Step for test it

### Setting for Github
---

- With your Github account, go to **Settings / Developer Setting / OAuth Apps**  and create an application
![0](https://user-images.githubusercontent.com/1847863/169177916-95d272a3-8f6e-4d4d-afc5-54d4161c0dbc.png)
- Set the **Authorization callback URL** the url of Adempiere ZK
![3](https://user-images.githubusercontent.com/1847863/169178241-52564bb7-269b-40d5-884d-5cc0e94076eb.png)
- Create an Application Registration with this characteristics
![1](https://user-images.githubusercontent.com/1847863/169178302-cb297cce-dfc2-4463-9a87-cbddd2c21dc5.png)
- Set the parameters with the information generated on Github (Client Identifier and Secret), in the Redirect URL set the url of Adempiere ZK
![2](https://user-images.githubusercontent.com/1847863/169178493-a3a8807b-dc3a-4a4f-926c-e22e1ccf377d.png)
 
### Setting for Google
---

- In your google console, go to **Credentials / Create Credentialas / Client ID Oauth** 
![4](https://user-images.githubusercontent.com/1847863/169179294-201be685-0e85-4928-8b16-5cff1a1e9247.png)
- Add a Redirect URI with the url of Adempiere ZK
- Create an Application Registration with this characteristics
![5](https://user-images.githubusercontent.com/1847863/169179523-38b1275e-f80b-4ce9-b551-40447101fdd9.png)
- Set the parameters with the information generated on Google (Client Identifier and Secret), in the Redirect URL set the url of Adempiere ZK
![6](https://user-images.githubusercontent.com/1847863/169179579-6658a020-59a6-4ebb-af59-25d435ca5527.png)

### Test functionality
---

- Logout and see the Adempiere Login 
![7](https://user-images.githubusercontent.com/1847863/169179690-5d3d7889-04ca-410a-a8cf-c290f2a9f76d.png)
- Select Login With Google and select an account or login with your google account
![8](https://user-images.githubusercontent.com/1847863/169179776-175bec8c-34e9-4d7a-9ba0-60cc5d5f131d.png)
- See the error message if not exists an login user registered with this email
![9](https://user-images.githubusercontent.com/1847863/169179963-3ab810ad-5262-40ee-8482-1ca69c226f39.png)
- Create an user with a valid email
![10](https://user-images.githubusercontent.com/1847863/169180088-c67351d7-2d39-4682-88e6-f2a347e0328b.png)
- Add a role for the user
![11](https://user-images.githubusercontent.com/1847863/169180335-c35e646d-5bc6-4930-9962-5266b07384fb.png)
- Logout and **Login With Google** and selected a valid email for Adempiere 
- After login this is redirect for Role selection panel
![12](https://user-images.githubusercontent.com/1847863/169180457-fb213082-656d-41a8-bd2a-ccdfc0bab647.png)
- Logout and **Login With Github** and selected a valid account for Adempiere 
![14](https://user-images.githubusercontent.com/1847863/169180699-8779e415-ca67-4000-8aec-98214ec06eee.png)
- After login this is redirect for Role selection panel
![12](https://user-images.githubusercontent.com/1847863/169180457-fb213082-656d-41a8-bd2a-ccdfc0bab647.png)
